src/profile-logic/processed-profile-versioning.js

```diff
@@ -2278,6 +2277,77 @@ const _upgraders = {
     // marker data with the new field types data, and no modification is needed in the
     // frontend to display older formats.
   },
+  [52]: (profile) => {
+    // This version simplifies how markers are mapped to their schema.
+    // The schema is now purely determined by data.type. The marker's name is ignored.
+    // If a marker has a null data, then it has no schema.
+    //
+    // In earlier versions, there were special cases for mapping markers with type
+    // "tracing" and "Text", and for markers with a null data property. These
+    // special cases have been removed.
+    //
+    // The upgrader for version 52 makes it so that existing profiles appear the
+    // same with the simplified logic. Specifically:
+    //  - Some old profiles have markers with data.type === 'tracing' but no schema
+    //    with the name 'tracing'. To ensure that the tracing markers from these
+    //    profile still show up in the 'timeline-overview' area, this upgrader adds
+    //    a schema to such profiles.
+    //  - Some old profiles have CC markers which only showed up in the memory track
+    //    because of special treatment of 'tracing' markers - the markers would have
+    //    data.type === 'tracing' and data.category === 'CC', and there would be a
+    //    'CC' schema with 'timeline-memory'. This upgrader moves these tracing CC
+    //    markers to a new 'tracingCCFrom52Upgrader' schema.
+    //
+    // Profiles from modern versions of Firefox already include a 'tracing' schema.
+    // And they don't use tracing markers for CC markers.
+
+    const schemaNames = new Set(profile.meta.markerSchema.map((s) => s.name));
+    const kTracingCCSchemaName = 'tracingCCFrom52Upgrader';
+    const shouldMigrateTracingCCMarkers =
+      schemaNames.has('CC') && !schemaNames.has(kTracingCCSchemaName);
```

Comment:
Hm. Is it possible to get `false` from `!schemaNames.has(kTracingCCSchemaName)`?

Reply:
Not realistically. I was just trying to be really careful to not mess with any information that was generated by non-Firefox sources. Though it's exceedingly unlikely that somebody happened to pick this exact name for their schema. I will remove it so that it's less confusing.